src/clarity/checkboxes/checkbox.ts

```diff
@@ -34,29 +35,54 @@ let latestId = 0;
 })
 export class Checkbox implements ControlValueAccessor {
     // If our host has an ID attribute, we use this instead of our index.
-    @Input("id") private _id: string = (latestId++).toString();
+    @Input("id")
+    private _id: string = (latestId++).toString();
+
     public get id() {
         return `clr-checkbox-${this._id}`;
     }
 
     // If our host has a name attribute, we apply it to the checkbox.
-    @Input("name") public name: string = null;
+    @Input("name")
+    public name: string = null;
 
     // Support for inline checkboxes, adds the necessary class to the host
     @Input("clrInline") public inline = false;
 
-    @Input("clrChecked") private _checked = false;
+    @Input("clrChecked")
+    private _checked = false;
+
     public get checked() {
         return this._checked;
     }
+
     public set checked(value: boolean) {
         if (value !== this._checked) {
             this._checked = value;
         }
     }
-    @Output("clrCheckedChange") public change = new EventEmitter<boolean>(false);
+
+    private _indeterminate: boolean = false;
+
+    public get indeterminate () {
+        return this._indeterminate;
+    }
+
+    @Input("clrIndeterminate")
+    public set indeterminate(value: boolean) {
+        this.checked = false;
+        this._indeterminate = value;
+        this.indeterminateChange.emit(this._indeterminate);
```

Comment:
So the expectation is that whenever a checkbox is set to indeterminate state that its checked value will be false?

We need a unit test to validate this.

Same for the other way, if a checkbox's checked state is set to false or true we need to make sure its indeterminate value is false.